mlflow/utils/nfs_on_spark.py

```diff
@@ -25,7 +29,18 @@ def get_nfs_cache_root_dir():
             == "true"
         )
         if nfs_enabled:
-            return "/local_disk0/.ephemeral_nfs"
+            nfs_root_dir = "/local_disk0/.ephemeral_nfs"
+            # Test whether the NFS directory is writable.
+            test_path = os.path.join(nfs_root_dir, uuid.uuid4().hex)
```

Comment:
Do we remove this line before the merge?

Reply:
Ah ok

Reply:
We'll keep this line because we're now testing whether or not we can write to NFS by attempting to create a directory at `test_path`

Reply:
Can we use `os.access(nfs_root_dir, os.W_OK)`?

ref: https://stackoverflow.com/a/2113511/6943581

Reply:
I think we should try to write, as is done in the current PR. The best way to test for write access is to try to write.

From the comments of https://stackoverflow.com/a/2113511/6943581:

```
Testing a directory for just the write bit isn't enough if you want to write files to the directory. You will need to test for the execute bit as well if you want to write into the directory. os.access('/path/to/folder', os.W_OK | os.X_OK) With os.W_OK by itself you can only delete the directory (and only if that directory is empty
```

If we're not careful, we might make a mistake.

Reply:
Sounds good!